oemof/outputlib/graph_tools.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+"""Modules for creating and manipulating energy system graphs."""
+
+import logging
+import warnings
+
+try:
+    from matplotlib import pyplot as plt
+except ImportError:
+    logging.warning('Matplotlib could not be imported.',
+                    ' Plotting will not work.')
```

Comment:
You should set `plt = None` in the except case. This is a common practice because it is more difficult to catch an undefined variable than asking if it is None. And it is much easier to find bugs.

As matplotlib is a requirement of oemof, the whole try/except statement might not be necessary.